server/chat-plugins/net-filters.ts

```diff
@@ -309,14 +320,35 @@ export const commands: ChatCommands = {
 			}
 			const targetPath = FS(`logs/chat/${roomid}/${date.slice(0, -3)}/${date}.txt`);
 			if (!targetPath.existsSync()) return this.errorReply(`Logs for that date not found`);
-			this.sendReply(`Initating training...`);
-			const response = await PM.query({data: {path: targetPath.path, result}, type: 'trainfrom'});
+			this.privateGlobalModAction(`${user.name} used /netfilter trainfrom ${roomid} (${date})`);
+			const response = await NeuralNetChecker.query({data: {path: targetPath.path, result}, type: 'trainfrom'}, true);
 			this.sendReply(`Training completed in ${Chat.toDurationString(response[0])}`);
 			this.privateGlobalModAction(
 				`${user.name} trained the net filters on logs from ${roomid} (${date} - ${Chat.count(response[1], 'lines')})`
 			);
 			this.stafflog(`Result: ${result} | Time: ${response[0]}ms`);
 		},
+		cancel: 'stop',
+		stop(target, room, user) {
+			checkAllowed(this);
+			let count = 0;
+			if (!PM.releasingProcesses.length) {
+				return this.errorReply(`No train tasks are pending`);
+			}
+			for (const subProcess of PM.releasingProcesses) {
+				for (const [task, resolve] of subProcess.pendingTasks) {
+					resolve({error: `The pending train query was cancelled.`});
+					subProcess.pendingTasks.delete(task);
+					count++;
+				}
+				// kill is done here to ensure the process is actually killed right now
+				// fsr the disconnect() done in destroy doesn't stop the running task
+				subProcess.process.kill();
+				subProcess.destroy();
```

Comment:
Does this actually clean the process up from `releasingProcesses` and `processes`?

Reply:
I believe so? I recall seeing it locally

Reply:
https://prnt.sc/yc7jpw

Reply:
Hm, this is probably a bit unintuitive, that `.destroy` will clean itself up from `.releasingProcesses` but not from `.processes`.

Reply:
I'm not sure if there's anything we can do about that, though.

Reply:
Actually, it's probable that `.process.disconnect` would handle removing it from `.processes`. So maybe this is fine overall! I should better comment `destroy` though.